source/agora/common/Deserializer.d

```diff
@@ -284,6 +284,14 @@ public void deserializePart (T) (
         foreach (const ref field; record.tupleof)
             deserializePart(field, dg);
 
+    // Support for associative arrays / hashmap
+    else static if (is(T : V[K], K, V))
+    {
+        size_t length = deserializeLength(dg, upperBound);
```

Comment:
Technically someone could create binary data with length = 1000, but where all the keys are the same. In the end we would have a hashmap with only one entry.

I can't really think of how this could be an attack vector. It's probably nothing to worry about..

Reply:
Good point, I didn't think of this. Yes it opens a few possibilities. Should we just disallow it ?

Reply:
something like `if (key in aa) throw`?

Reply:
It's a double lookup, so I was using of using `update` https://github.com/dlang/druntime/blob/9354a9207f4417d4b97c42f4a19dd93c2bfe3108/src/object.d#L2650